agent/engine/docker_image_manager.go

```diff
@@ -267,8 +271,12 @@ func (imageManager *dockerImageManager) removeExistingImageNameOfDifferentID(con
 }
 
 func (imageManager *dockerImageManager) StartImageCleanupProcess(ctx context.Context) {
-	// passing the cleanup interval as argument which would help during testing
-	imageManager.performPeriodicImageCleanup(ctx, imageManager.imageCleanupTimeInterval)
+	// If the image pull behavior is prefer cached, don't clean up the image,
+	// because the cached image is needed.
+	if imageManager.imagePullBehavior != config.ImagePullPreferCachedBehavior {
+		// passing the cleanup interval as argument which would help during testing
```

Comment:
[do this!]
log that you are disabling this.

[nit]
I think I'd prefer positive comparison instead of !=.
```
if imageManager.imagePullBehavior == config.ImagePullPreferCachedBehavior {
    seelog.Info("Pull behavior is set to always use cache. Disabling cleanup")
    return
}
imageManager.performPeriodicImageCleanup(ctx, imageManager.imageCleanupTimeInterval) 
```

Reply:
OK, sure.